consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java

```diff
@@ -260,38 +268,80 @@ public void run() {
             getJobExecutions().add(currentRun);
 
             configureClients();
+            int jobExecutionPsn = -1;
 
-            saveJobDetails(true);
-
-            Map<String, JsonNode> sourceDocuments = getSourceDocuments();
+            LightblueResponse response = saveJobDetails(jobExecutionPsn);
 
-            Map<String, JsonNode> destinationDocuments = getDestinationDocuments(sourceDocuments);
+            boolean processJob = true;
 
-            List<JsonNode> documentsToOverwrite = getDocumentsToOverwrite(sourceDocuments, destinationDocuments);
+            // verify this job is the first active thread processing the thread.
+            // first extract processed data as MigrationJob
+            JsonNode node = response.getJson().path("processed");
+            try {
```

Comment:
Should we pull out the processJob (between here and line 311) logic into a separate method?